### DISCUSSION TITLE
[Feature Request] Native Elementor A/B testing/Split Testing that is performance optimized

### DISCUSSION THREAD

**Discussion by jazir555** [elementor/elementor#24625] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

It would be awesome if Elementor had a natively integrated A/B testing feature.  There is currently no A/B testing feature that works with all Elementor elements(as well as Elementor Add-ons) that allows designers to create multiple versions of a page and test them against each other to see which one visitors prefer. 

Adding this feature would allow us to compare the traffic between each page, as well as which elements were interacted with most,which would help to compare(for example) orange vs blue buttons to see which converts better.

[Divi Leads](https://www.elegantthemes.com/blog/theme-releases/divi-leads-for-the-visual-builder) is somewhat comparable to what this request is for.

### Describe the solution you'd like.

Natively integrated A/B Testing/Split Testing of multiple designs of the same page at the Same URL. The product page for instance. A user could visit a product page and 50% of users receive one design and 50% of users receive the other, and there would be tracked metrics associated with each page and each element contained on the page. 

Since Split Testing would potentially add some additional latency for pages on which the feature was enabled, it would be ideal if the feature could be activated on a page by page basis to ensure that pages that are not being split tested have no additional code loaded(so it would be enabled selectively).

Additionally, the feature would be compatible with caching and other performance optimization techniques(remove unused css, delaying javascript, deferring javascript, minification of JS and CSS, etc). No A/B plugins on the market allow for optimization compatibility.

### Describe alternatives you've considered.

https://wordpress.org/plugins/split-test-for-elementor/#description

Split test for Elementor is extremely limited. As are Nelio Unlocker, Split Hero etc. All of them have horrific performance issues as they conflict with caching and other optimization solutions, leaving the A/B testing pages with absolutely terrible performance since they are entirely unoptimized.

Divileads is a good example, but Divi is an absolutely terrible builder.

### Additional context

All A/B/Split testing plugins have very limited utility and _all of them_ have terrible performance issues since they are incompatible with most optimization techniques. Natively building this feature into Elementor with a focus on performance optimization would eliminate the need for 3rd party plugins, and bring full compatibility with all elements and addons for Elementor directly integrated within Elementor.

No more third party plugins, no more jank. 

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by eaglevisionsolutions** [comment]
I second this feature too this would be very helpful and a very powerful tool.

**Comment by StenderIternum** [comment]
Yes please - Right now we are doing split testing by using request parameters to determine which containers to load, but this also makes it so that cache do not work on the splits.
